### ISSUE TITLE
STATUSMSG interferes with hilights

### ISSUE THREAD

**Issue by ilbelkyr** [irssi/irssi#354]
Some brief testing:

```
1114 104804 <+Nymeria> normal message test
1114 104838 <+Nymeria> hilight ilbelkyr normal message test
1114 104902 <+Nymeria:+#apastebin> statusmsg test
1114 104929 <+Nymeria:+#apastebin> should hilight ilbelkyr but doesn't! statusmsg test
```

The 2nd and 4th of these messages should be hilighted, but only the 2nd is; the others are only considered normal non-hilight activity. (According to `printlevels.pl`, all of these are `PUBLICS`; only the 2nd is also `HILIGHTS`.)

Relevant hilight entry: `1114 103840    2 ilbelkyr  -priority 50`

Possibly relevant `set hilight` output:

```
1114 103845 [lookandfeel]
1114 103845 activity_hilight_level = MSGS DCCMSGS
1114 103845 hilight_act_color = %M
1114 103845 hilight_color = %Y
1114 103845 hilight_level = PUBLIC DCCMSGS NOTICES MODES WALLOPS MSGS TOPICS
1114 103845 hilight_nick_matches = OFF
1114 103845 hilight_nick_matches_everywhere = OFF
```

Notices behave as expected:

```
1114 110858 |NOTICES| -Nymeria:#apastebin- notice
1114 110905 |NOTICES HILIGHTS| -Nymeria:#apastebin- notice ilbelkyr
1114 110925 |NOTICES| -Nymeria:+#apastebin- statusnotice
1114 110928 |NOTICES HILIGHTS| -Nymeria:+#apastebin- statusnotice ilbelkyr
```

**Comment by vague666** [comment]
If you try
1114 103845 hilight_nick_matches = ON
1114 103845 hilight_nick_matches_everywhere = ON
does it work better? (remove the hilight for the test)

**Comment by ilbelkyr** [comment]
Doesn't work for me with those, either. :(

**Comment by dequis** [comment]
>10:16 < vague> dx, any idea about this, https://github.com/irssi/irssi/issues/354 ?
>10:19 < dx> all this stuff https://github.com/irssi/irssi/blob/ac04fa7/src/fe-common/core/fe-messages.c#L185-L215
>10:19 < dx> ...is missing from https://github.com/irssi/irssi/blob/ac04fa7/src/fe-common/irc/fe-irc-messages.c#L70-L96

**Comment by ilbelkyr** [comment]
I hate to be the annoying person poking about open bugs, but given that [freenode is planning to use STATUSMSG for messages affected by `+z`][fnblog], I'd personally quite appreciate this being fixed.

It does look like @dequis identified the issue, although it feels like the amount of repeated code between the varying codepaths might be another problem :(

[fnblog]: https://freenode.net/news/seven-118

**Comment by ailin-nemui** [comment]
@ilbelkyr can you see if this is fixed by #1129
